ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateSignatureProvider.fs

```diff
@@ -171,15 +171,18 @@ type FSharpGenerateSignatureBuilder() =
                 let sourceString (binding: IBinding) =
                     let sb = StringBuilder()
 
-                    sb.Append("val ") |> ignore
-                    sb.Append(binding.HeadPattern.GetText()) |> ignore
-                    sb.Append(": ") |> ignore
-                    
                     let refPat = binding.HeadPattern.As<IReferencePat>()
                     if isNotNull refPat then
                         let symbolUse = refPat.GetFcsSymbolUse()
                         if isNotNull symbolUse then
                             let mfv = symbolUse.Symbol :?> FSharpMemberOrFunctionOrValue
+
+                            mfv.Attributes
+                            |> Seq.iter (fun a -> sb.Append($"{a.Format(symbolUse.DisplayContext)}{lineEnding}") |> ignore)
```

Comment:
What happens when the attribute has arguments?
`[<Ignore "Foobar">]`, I'd be surprised if the Format takes care of that.
Does the factory generated member have a `SetAttribute` call instead?

Reply:
I adjusted the test. I think Format is doing what we need regarding arguments.

Reply:
Until it won't, try
```fsharp
module Telplin

type MyEnum =
    | A = 0
    | B = 1

type BAttribute(v:MyEnum) =
    inherit System.Attribute()

[<BAttribute(MyEnum.A)>]
let a b c = b + c
```

I think this won't work out.
The safer option would be to reuse the attribute from the implementation file.

Reply:
How does it work with named literal values?

```fsharp
[<Flags>]
type Enum =
    | None = 0
    | Aa = 1
    | Bb = 2
    | Cc = 4
    | AaAndCc = 5

type AttrAttribute() =
    inherit Attribute()

    member val Prop = Enum.None with get, set 

[<Attr(Prop = Enum.AaAndCc)>]
let a = 1
```

```fsharp
let [<Literal>] Hello = "Hello"

[<CompiledName(Hello)>]
let hello = 1
```

Reply:
Yup, https://github.com/dotnet/fsharp/issues/13810

Reply:
> Yup, https://github.com/dotnet/fsharp/issues/13810

IIRC FCS doesn't preserve the reference to a named literal and only captures the literal value. So if you have multiple literals with the same value, it wouldn't know which one to use.

Consider this case:
```
let [<Literal>] Hello1 = "Hello"
let [<Literal>] Hello2 = "Hello"

[<CompiledName(Hello1)>]
let hello = 1
```

We want to preserve the named literal reference, so it's possible to update the attribute values in both signature and implementation files by changing the single named literal value. And it'll also prevent features like Find Usages and Rename from breaking.

> The safer option would be to reuse the attribute from the implementation file.

This looks like the way to go, yes.

Reply:
Hehe, you guys know your stuff.
So I used `GetText()` to get the original attribute from the implementation file. Is that okay?
I still need to make `IEnumRepresentation` work...

Reply:
This is not what I meant. So, `binding` has a bunch of setters to insert existing nodes into it.
Like
![image](https://github.com/JetBrains/resharper-fsharp/assets/2621499/57ab74bd-42d9-4c45-987c-19a5635b082a)


And wishful thinking the newly created `IModuleMember` had something similar like `SetAttribute`
@auduchinok any ideas?

Reply:
> And wishful thinking the newly created IModuleMember had something similar like SetAttribute

For some reason the generator didn't create it. I'd suggest to work with tree manually until we can enable the formatter to handle all the whitespace-related modifications for us.

Reply:
By just inserting the nodes before the val node?
![image](https://github.com/JetBrains/resharper-fsharp/assets/2621499/ebdd5b9f-fa4b-4f23-8675-25b9a4daeebd)